lib/widgets/twake_components/twake_preview_link/twake_link_view.dart

```diff
@@ -0,0 +1,151 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/app_state/preview_url/get_preview_url_success.dart';
+import 'package:fluffychat/pages/chat/events/message/message_style.dart';
+import 'package:fluffychat/presentation/extensions/media/url_preview_extension.dart';
+import 'package:fluffychat/widgets/clean_rich_text.dart';
+import 'package:fluffychat/widgets/mxc_image.dart';
+import 'package:fluffychat/widgets/twake_components/twake_preview_link/twake_link_view_style.dart';
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/linagora_design_flutter.dart';
+import 'package:matrix_link_text/link_text.dart';
+
+class TwakeLinkView extends StatelessWidget {
+  final String text;
+  final Widget childWidget;
+  final bool ownMessage;
+  final TextStyle? textStyle;
+  final TextStyle? linkStyle;
+  final TextAlign? textAlign;
+  final LinkTapHandler? onLinkTap;
+  final int? maxLines;
+  final String? firstValidUrl;
+  final TextSpanBuilder? textSpanBuilder;
+  final ValueNotifier<Either<Failure, Success>> getPreviewUrlStateNotifier;
```

Comment:
why `stateless` still need `ValueNotifier`?

Reply:
Inside the class, I want to rebuild a small widget based on ValueNotifier. I think it has nothing to do with `stl`

Reply:
It makes the `stateless` has `state` :)

Reply:
I deleted it